main.go

```diff
@@ -66,6 +72,7 @@ func verify(ctx context.Context,
 }
 
 func main() {
+	flag.StringVar(&builderID, "builder-id", "", "EXPERIMENTAL: the unique builder ID")
```

Comment:
nit: "unique builder ID specified in the provenance"? 

is this not a provenance options?

Reply:
it is in both the certificate and the provenance for our builders. I can move it to provenanceOpts. Wdut?

Reply:
Changed text to `the unique builder ID who created the provenance`.
Let me know if I should get rid f the builderOpt and use the provenanceOpt instead. You mentioned buildOpt offline, so I took this as a hint ^^. Can change it

Reply:
Ah, after reading the code I think we should keep in both. Can we add a BuilderID in the provenance opts and use this flag to populate both the field in the builderopts and the provenanceopts? 

that way `VerifyProvenance` only takes provenance opts. Same with `VerifyWorkflowIdentity`: this should only consume from the builder options I think. do you agree?

Reply:
That's fine with me. The `VerifyWorkflowIdentity` still takes a source as well, but we can fix it in a follow-up PR.

Reply:
awesome, plan could be that we do a ConstructBuilderOptsFor(GHA/GCB), ConstructProvenanceOptsFor(GHA/GCB). Follow-up sg.